sheeprl/algos/dreamer_v2/utils.py

```diff
@@ -0,0 +1,276 @@
+                raise ValueError(
+                    f"Negative value of atart_noop_max parameter ({args.atari_noop_max}), the minimum value allowed is 0"
+                )
+            env = gym.wrappers.AtariPreprocessing(
+                env,
+                noop_max=args.atari_noop_max,
+                frame_skip=args.action_repeat,
+                screen_size=64,
+                grayscale_obs=args.grayscale_obs,
+                scale_obs=False,
+                terminal_on_life_loss=False,
+                grayscale_newaxis=True,
+            )
+            env = gym.wrappers.TimeLimit(env, 27000)
+        else:
+            env = ActionRepeat(env, args.action_repeat)
+            if isinstance(env.observation_space, gym.spaces.Box) or len(env.observation_space.shape) < 3:
+                env = gym.wrappers.PixelObservationWrapper(env)
+                env = gym.wrappers.TransformObservation(env, lambda obs: obs["pixels"])
+                env.observation_space = env.observation_space["pixels"]
+            env = gym.wrappers.ResizeObservation(env, (64, 64))
+            if args.grayscale_obs:
+                env = gym.wrappers.GrayScaleObservation(env, keep_dim=True)
+        env = gym.wrappers.TransformObservation(env, lambda obs: obs.transpose(2, 0, 1))
+        env.observation_space = gym.spaces.Box(
+            0, 255, (env.observation_space.shape[-1], *env.observation_space.shape[:2]), np.uint8
+        )
+    env.action_space.seed(seed)
+    env.observation_space.seed(seed)
+    if args.max_episode_steps > 0:
+        env = gym.wrappers.TimeLimit(env, max_episode_steps=args.max_episode_steps // args.action_repeat)
+    env = gym.wrappers.RecordEpisodeStatistics(env)
+    if args.capture_video and rank == 0 and run_name is not None:
+        env = gym.experimental.wrappers.RecordVideoV0(
+            env, os.path.join(run_name, prefix + "_videos" if prefix else "videos"), disable_logger=True
+        )
+        env.metadata["render_fps"] = env.frames_per_sec
+    return env
+
+
+def compute_stochastic_state(
+    logits: Tensor,
+    event_shape: Optional[int] = 1,
+    discrete: int = 32,
+) -> Tensor:
+    """
+    Compute the stochastic state from the information of the distribution of the stochastic state.
+
+    Args:
+        logits (Tensor): logits from either the representation model or the transition model.
+        event_shape (int, optional): how many batch dimensions have to be reinterpreted as event dims.
+            Default to 1.
+        discrete (int, optional): the size of the Categorical variables.
+            Defaults to 32.
+
+    Returns:
+        The mean and the standard deviation of the distribution of the stochastic state.
+        The sampled stochastic state.
+    """
+    logits = logits.view(*logits.shape[:-1], -1, discrete)
+    dist = Independent(OneHotCategoricalStraightThrough(logits=logits), event_shape)
+    return dist.rsample()
+
+
+def compute_lambda_values(
```

Comment:
can't this be needed for other algorithms too? maybe to be placed as global utils instead of dreamer's ones?